scripts/bootloader/zb_add_ota_header.py

```diff
@@ -0,0 +1,204 @@
+    parser.add_argument("--zigbee-image-type", required=False, type=auto_int,
+                        help="Image type to be used in Zigbee OTA header.")
+    parser.add_argument("--zigbee-comment", required=False, nargs='?',
+                        help="Firmware comment to be used in Zigbee OTA header.")
+    parser.add_argument("--zigbee-ota-min-hw-version", required=False, type=auto_int, nargs='?',
+                        help="The zigbee OTA minimum hw version of Zigbee OTA Client.")
+    parser.add_argument("--zigbee-ota-max-hw-version", required=False, type=auto_int, nargs='?',
+                        help="The zigbee OTA maximum hw version of Zigbee OTA Client.")
+    return parser.parse_args()
+
+
+def main():
+
+    args = parse_args()
+
+    app_file = args.application
+    app_version_int = convert_version_string_to_int(args.application_version_string)
+    zb_manufacturer_id = args.zigbee_manufacturer_id if args.zigbee_manufacturer_id is not None else OTA_HEADER_MANUFACTURER_WILDCARD
+    zb_image_type = args.zigbee_image_type if args.zigbee_image_type is not None else OTA_HEADER_IMAGE_TYPE_WILDCARD
+
+    if args.zigbee_comment is None:
+        zb_comment = ''
+    elif any(ord(char) > 127 for char in args.zigbee_comment): # Check if all the characters belong to the ASCII range
+        print('Warning: Non-ASCII characters in the comment are not allowed. Discarding comment.')
+        zb_comment = ''
+    elif len(args.zigbee_comment) > 30:
+        print('Warning: truncating the comment to 30 bytes.')
+        zb_comment = args.zigbee_comment[:30]
+
+    if args.zigbee_ota_min_hw_version is not None and args.zigbee_ota_min_hw_version > 0xFFFF:
+        raise print('zigbee-ota-min-hw-version parameter exceeds 2-byte long integer.')
+    zb_ota_min_hw_version = args.zigbee_ota_min_hw_version
+
+    if args.zigbee_ota_max_hw_version is not None and args.zigbee_ota_max_hw_version > 0xFFFF:
+        raise print('zigbee-ota-max-hw-version parameter exceeds 2-byte long integer.')
+    zb_ota_max_hw_version = args.zigbee_ota_max_hw_version
+
+    # Warn user if minimal/maximum zigbee ota hardware version are not correct:
+    #   * only one of them is given
+    #   * minimum version is higher than maximum version
+    #   * hw_version is inside the range specified by minimum and maximum hardware version
+    if isinstance(zb_ota_min_hw_version, int) != isinstance(zb_ota_max_hw_version, int):
+        print('Warning: min/max zigbee ota hardware version is missing. Discarding min/max hardware version.')
+    elif isinstance(zb_ota_min_hw_version, int):
+        if zb_ota_min_hw_version > zb_ota_max_hw_version:
+            print('Warning: zigbee-ota-min-hw-version is higher than zigbee-ota-max-hw-version.')
+
+
+    zigbee_ota_file = OTA_file(app_version_int,
+                                os.path.getsize(app_file),
+                                bytes(open(app_file, 'rb').read()),
```

Comment:
I'd read the file before constructing the OTA file object, with the appropriate error-handling.